packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,295 @@
+
+        if (msg.sender != safeConfigs[_safe].fallbackOwner) {
+            revert LivenessModule2_UnauthorizedCaller();
+        }
+
+        if (challengeStartTime[_safe] != 0) {
+            revert LivenessModule2_ChallengeAlreadyExists();
+        }
+
+        challengeStartTime[_safe] = block.timestamp;
+        emit ChallengeStarted(_safe, block.timestamp);
+    }
+
+    /// @notice Responds to a challenge for an enabled safe, canceling it
+    /// @dev MUST only be executable by an enabled safe
+    /// @dev MUST revert if there isn't a challenge for the calling safe
+    /// @dev MUST revert if challenge exists but response period has expired
+    /// @dev MUST emit the ChallengeCancelled event
+    function respond() external {
+        // Check that this module is enabled on the calling Safe
+        _assertModuleEnabled(msg.sender);
+
+        // Check if the calling safe has configuration set
+        _assertModuleConfigured(msg.sender);
+
+        uint256 startTime = challengeStartTime[msg.sender];
+        if (startTime == 0) {
+            revert LivenessModule2_ChallengeDoesNotExist();
+        }
+
+        // Check if response period has expired
+        if (block.timestamp >= getChallengePeriodEnd(msg.sender)) {
+            revert LivenessModule2_ResponsePeriodEnded();
+        }
+
+        _cancelChallenge(msg.sender);
+    }
+
+    /// @notice With successful challenge, removes all current owners from enabled safe,
+    /// sole owner, and sets its quorum to 1
+    /// @dev MUST be executable by anyone
+    /// @dev MUST revert if the given safe hasn't enabled the module
+    /// @dev MUST revert if there isn't a successful challenge for the given safe
+    /// @dev MUST enable the module to start a new challenge
+    /// @dev MUST emit the ChallengeSucceeded event
+    /// @param _safe The Safe to transfer ownership of
+    function changeOwnershipToFallback(address _safe) external {
+        // Ensure Safe is configured with this module to prevent unauthorized execution
+        _assertModuleConfigured(_safe);
+
+        // Verify module is still enabled to ensure Safe hasn't disabled it mid-challenge
+        _assertModuleEnabled(_safe);
+
+        // Verify active challenge exists - without challenge, ownership transfer not allowed
+        uint256 startTime = challengeStartTime[_safe];
+        if (startTime == 0) {
+            revert LivenessModule2_ChallengeDoesNotExist();
+        }
+
+        // Ensure response period has fully expired before allowing ownership transfer.
+        // This gives Safe owners full configured time to demonstrate liveness.
+        if (block.timestamp < getChallengePeriodEnd(_safe)) {
+            revert LivenessModule2_ResponsePeriodActive();
+        }
+
+        Safe targetSafe = Safe(payable(_safe));
+
+        // Get current owners
+        address[] memory owners = targetSafe.getOwners();
+
+        // Remove all owners after the first one
+        while (owners.length > 1) {
```

Comment:
In theory this runs out of gas if there are a lot of owners, but whatever. An auditor would likely report this as a low-severity issue. You'd be able to prevent this by preventing the module from being used for any safes where the owner count > some threshold limit of owners.

Reply:
That owner count would probably need to be in the hundreds or thousands so unlikely to be an issue in practice